docs/proposals/controller/api_definition.yaml

```diff
@@ -0,0 +1,177 @@
+openapi: 3.0.0
+info:
+  title: Cluster Controller API
+  description: External API to send requests to the controller.
+  version: 0.1.0
+servers:
+- url: http://<api-link>/v0
+  description: Main API server
+paths:
+  /workload:
+    get:
+      summary: Returns a list of workloads.
+      responses:
+        '200':
+          description: A JSON array of workload description objects
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Workload'
+    post:
+      summary: Creates a workload (only container workload type for now)
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Workload'
+      responses:
+        '201':
+          description: Created
+        '400':
+          description: Yaml description file is invalid
+  /workload/{id}:
+    get:
+      summary: Returns a single workload by id.
+      responses:
+        '200':
+          description: A workload description file
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Workload'
+    delete:
+      summary: Deletes a workload by id
+      responses:
+        '204':
+          description: Deleted
+    parameters:
+    - name: id
+      in: path
+      description: ID of a workload
+      required: true
+      schema:
+        type: string
+  /instance:
+    get:
+      summary: Returns a list of all instances.
+      responses:
+        '200':
+          description: A JSON array of istance state objects
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/ShortState'
```

Comment:
Just return the list of instance ids here, without state.

Reply:
Done